tests/mock_callable_testslide.py

```diff
@@ -101,6 +108,28 @@ def patching_private_functions_with_allow_private(self):
         ).and_assert_called_once()
         t._privatefun()
 
+    @context.example
+    def return_value_raises_with_coroutine(self):
+        with self.assertRaises(ValueError):
+            self.mock_callable(
+                sample_module, "test_function", type_validation=False
+            ).to_return_value(coro_fun())
+
+    @context.example
+    def return_values_raises_with_coroutine(self):
+        with self.assertRaises(ValueError):
+            self.mock_callable(
+                sample_module, "test_function", type_validation=False
+            ).to_return_values([1, 2, coro_fun()])
+
+    @context.example
+    def with_implementation_raises_with_coroutine(self):
+        with self.assertRaises(ValueError):
+            self.mock_callable(
+                sample_module, "test_function", type_validation=False
+            ).with_implementation(coro_fun)
+            sample_module.test_function("a", "b")
+
```

Comment:
These tests are at the top level, without any specific context. I reckon these tests aren't easy to maintain...

If you fold the sections:

![image](https://user-images.githubusercontent.com/1386232/91028229-14fb8e80-e5f5-11ea-8d26-889bf742e4ed.png)

it gets possible to see each use case:

- to_return_value
- to_return_values
- to_yield_values
- with_implementation
- with_wrapper

I think we can move each of the 3 examples above inside these sections and add example for the 2 missing cases. Historically, having these "test all scenarios / cases" tests helped catch a lot of bugs.

I know these tests are pretty bad to maintain, I won't feel offended you tell me to do it myself, lol.